### PR TITLE
Douyutv fix issue #1102 #1004

### PR DESCRIPTION
**Pull request by fozzysec**
fix for issue #1102 and issue #1004
Thanks for new API from https://github.com/spacemeowx2/DouyuHTML5Player/issues/28 by @ERioK
Also has an alternative way to get HLS stream by @back-to at https://gist.github.com/back-to/9cec89ecc9c1fb7556523a8658391375

### CODE REVIEW

**Comment by steven7851** [comment]
There are alternative quality for your refrence.
https://github.com/steven7851/streamlink/blob/patch-14/src/streamlink/plugins/douyutv.py

File: src/streamlink/plugins/douyutv.py [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
+API_URL = "https://capi.douyucdn.cn/api/v1/{0}&auth={1}"
 VAPI_URL = "https://vmobile.douyu.com/video/getInfo?vid={0}"
-LAPI_SECRET = "a2053899224e8a92974c729dceed1cc99b3d8282"
+API_SECRET = "Y237pxTx2In5ayGz"
 SHOW_STATUS_ONLINE = 1
 SHOW_STATUS_OFFLINE = 2
 STREAM_WEIGHTS = {
-    "low": 540,
-    "medium": 720,
+    "middle": 720,
```
Comment (gravyboat) [inline src/streamlink/plugins/douyutv.py]: Is this seriously named middle? Same thing for line 18 at middle2?

Comment (fozzysec) [inline src/streamlink/plugins/douyutv.py]: The name "middle" and "middle2" are returned by the json from the API, of course we can rename it to medium and medium2.

**Comment by fozzysec** [comment]
Here is the json sample returned by the API
```
{
    "data": {
        "anchor_city": "\u6210\u90fd\u5e02",
        "black": [],
        "cate_id": "174",
        "cate_id1": 2,
        "cdns": [
            "ws",
            "tct",
            "ws2"
        ],
        "cdnsWithName": [
            {
                "cdn": "ws",
                "name": "\u4e3b\u7ebf\u8def"
            },
            {
                "cdn": "tct",
                "name": "\u5907\u7528\u7ebf\u8def5"
            },
            {
                "cdn": "ws2",
                "name": "\u5907\u7528\u7ebf\u8def2"
            }
        ],
        "credit_illegal": "0",
        "cur_credit": "12",
        "fans": "369281",
        "game_icon_url": "https://staticlive.douyucdn.cn/upload/game_cate/e6aa94fa2d6c5498bd72f11422710f3e.png",
        "game_name": "\u4e8c\u6b21\u5143",
        "game_url": "/directory/game/ecy",
        "ggad": {
            "play1": "",
            "play2": "",
            "play3": "",
            "play4": "",
            "play5": "",
            "videop": ""
        },
        "gift": [
            {
                "big_effect_icon": "https://staticlive.douyucdn.cn/upload/dygift/1704/d882f05a34e115496554e2d030f3e19b.gif",
                "brgb": "#ff831f",
                "ch": "0",
                "cimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/1fd123f74dab9489e92b93d081f0777b.gif",
                "desc": "\u8d60\u9001\u7f51\u7ad9\u5e7f\u64ad\u5e76\u6d3e\u9001\u51fa\u795e\u79d8\u5b9d\u7bb1",
                "drgb": "#210101",
                "ef": 1,
                "effect": "143",
                "grgb": "#a20b0b",
                "gt": "2",
                "gx": 5000,
                "himg": "https://staticlive.douyucdn.cn/upload/dygift/1707/ee92c8c08dd4cc3ea6d67eaae47b69ed.gif",
                "id": "988",
                "intro": "\u6211\u4eec\u7684\u5f81\u9014\u662f\u661f\u8fb0\u5927\u6d77",
                "m_ef_gif_1": "https://staticlive.douyucdn.cn/upload/dygift/1704/07b3fcfa0adfc7b0b325e0dbdf133ae6.gif",
                "m_ef_gif_2": "",
                "mgif": "https://staticlive.douyucdn.cn/upload/dygift/1707/6b49bbeb3ea4d8b40f2c71d0943719a1.gif",
                "mimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/6149357aeb48979f91cca9f1de349d3b.png",
                "mobile_big_effect_icon_0": "https://staticlive.douyucdn.cn/upload/dygift/1704/07b3fcfa0adfc7b0b325e0dbdf133ae6.gif",
                "mobile_big_effect_icon_1": "https://staticlive.douyucdn.cn/upload/dygift/1704/b75d7f9ca9fcbbf9d232c77d3a32d0cd.png",
                "mobile_big_effect_icon_2": "https://staticlive.douyucdn.cn/upload/dygift/1704/e2fe95234b5532ec88413f9b546d12fb.png",
                "mobile_big_effect_icon_3": "https://staticlive.douyucdn.cn/upload/dygift/1704/14a51212617c733d49f392f4357a99c6.png",
                "mobile_icon_v2": "https://staticlive.douyucdn.cn/upload/dygift/1707/82de32fc43ccdbf9cbf963aa818aec77.png",
                "mobile_small_effect_icon": "https://staticlive.douyucdn.cn/upload/dygift/1704/a05949926b89217a42b6c009f5ad9447.png",
                "mobile_stay_time": "150000",
                "mobimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/f7c2040ca2442e269889784dca88bb38.png",
                "name": "\u732b\u8033\u706b\u7bad",
                "pad_big_effect_icon": "https://staticlive.douyucdn.cn/upload/dygift/1704/8897a5f6b2ab6d5676621f7b2bb84885.gif",
                "pc": "50000",
                "pdbimg": "",
                "pdhimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/6fb90eedc2686a7e8e5113c5277275e9.gif",
                "pimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/ad0fbf0f90c354b004d2046146976ab5.png",
                "pt": "\u4f59\u989d\u4e0d\u8db3\uff0c\u706b\u7bad\u88ab\u5361\u5728\u53d1\u5c04\u5854\u4e0a\u4e86",
                "small_effect_icon": "https://staticlive.douyucdn.cn/upload/dygift/1707/ac3ee14c59fc30d0bca1795d554babce.png",
                "sort": "6",
                "stay_time": 200000,
                "type": "2",
                "urgb": "#750101"
            },
            {
                "big_effect_icon": "https://staticlive.douyucdn.cn/upload/dygift/1609/cc88750512f6759d02c157e80322df90.png",
                "brgb": "#9a76c5",
                "ch": "0",
                "cimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/3ef88917e109e6e54667098af5c03d67.gif",
                "desc": "\u8d60\u9001\u623f\u95f4\u5e7f\u64ad\u5e76\u8d60\u9001\u9053\u5177\u5b9d\u7bb1",
                "drgb": "#210101",
                "ef": 2,
                "effect": "63",
                "grgb": "#9104af",
                "gt": "0",
                "gx": 1000,
                "himg": "https://staticlive.douyucdn.cn/upload/dygift/1606/93daef170894a9d6bd8495fa0f81e165.gif",
                "id": "195",
                "intro": "\u4e3b\u64ad\u5e26\u6211\u98de",
                "m_ef_gif_1": "https://staticlive.douyucdn.cn/upload/dygift/1609/ff4ca2b702b6d93f3add303aff21fb0e.gif",
                "m_ef_gif_2": "",
                "mgif": "",
                "mimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/76dd0d8c803b133467e4f0196c7d071d.png",
                "mobile_big_effect_icon_0": "https://staticlive.douyucdn.cn/upload/dygift/1609/ff4ca2b702b6d93f3add303aff21fb0e.gif",
                "mobile_big_effect_icon_1": "https://staticlive.douyucdn.cn/upload/dygift/1609/729d1f1c0ca9a2f5db13d6f4ddf4514f.png",
                "mobile_big_effect_icon_2": "https://staticlive.douyucdn.cn/upload/dygift/1609/8ef4e9ce4b2b46b7d80ffa4b598599df.png",
                "mobile_big_effect_icon_3": "https://staticlive.douyucdn.cn/upload/dygift/1609/8ce7b8e97762ac751d67b2ee34aa79b4.png",
                "mobile_icon_v2": "https://staticlive.douyucdn.cn/upload/dygift/1606/da5a1fd20028953ad93a8a2bbd359a89.png",
                "mobile_small_effect_icon": "",
                "mobile_stay_time": "30000",
                "mobimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/ac245d735b62dcae65dd3eea9352b7e3.png",
                "name": "\u98de\u673a",
                "pad_big_effect_icon": "https://staticlive.douyucdn.cn/upload/dygift/1609/a98bfe51d035849f79774af8798da2ba.png",
                "pc": "10000",
                "pdbimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/f7a44c0b6ac21210e905447929e8de39.png",
                "pdhimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/f0ae1148d092f0cb2fe162152e4bff16.gif",
                "pimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/d8993d42da06ae6c625ace3dff6448a1.png",
                "pt": "\u4f59\u989d\u4e0d\u8db3\uff0c\u98de\u673a\u8fd8\u5728\u6ed1\u884c\u4e2d",
                "small_effect_icon": "https://staticlive.douyucdn.cn/upload/dygift/1609/03814221b8af7257d0960a3b4bcf4fd8.png",
                "sort": "5",
                "stay_time": 30000,
                "type": "2",
                "urgb": "#581c86"
            },
            {
                "big_effect_icon": "",
                "brgb": "#ff850d",
                "ch": "0",
                "cimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/5c46a60e26298d2830ec7bf6b9fb6de2.gif",
                "desc": "",
                "drgb": "#ff850d",
                "ef": 0,
                "effect": "0",
                "grgb": "#ff850d",
                "gt": "0",
                "gx": 60,
                "himg": "https://staticlive.douyucdn.cn/upload/dygift/1704/130dc639eadd4f79b3d9e24ead4b9ced.gif",
                "id": "750",
                "intro": "\u8001\u677f\uff0c\u6765\u529e\u5f20\u5361\u5427\uff01",
                "m_ef_gif_1": "",
                "m_ef_gif_2": "",
                "mgif": "https://staticlive.douyucdn.cn/upload/dygift/1704/fc9e61793a53a383887183468f003a86.gif",
                "mimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/74c75c59a1e9338d0b25b62546b84b1d.png",
                "mobile_big_effect_icon_0": "",
                "mobile_big_effect_icon_1": "",
                "mobile_big_effect_icon_2": "",
                "mobile_big_effect_icon_3": "",
                "mobile_icon_v2": "https://staticlive.douyucdn.cn/upload/dygift/1704/6277a93275fffa08635790677f01e913.png",
                "mobile_small_effect_icon": "",
                "mobile_stay_time": "6000",
                "mobimg": "",
                "name": "\u529e\u5361",
                "pad_big_effect_icon": "",
                "pc": "600",
                "pdbimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/2e15e51eacbd9f5f2c09e9462a29cbc2.png",
                "pdhimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/725f9633c5e66426e6b3a91485c9e17e.gif",
                "pimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/655e318cd934a0a9d20f86c5fa92cdf8.png",
                "pt": "",
                "small_effect_icon": "",
                "sort": "4",
                "stay_time": 6000,
                "type": "2",
                "urgb": "#ff850d"
            },
            {
                "big_effect_icon": "",
                "brgb": "#ff4f81",
                "ch": "0",
                "cimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/112687d87194ab09fd739ea07eaf7c0f.gif",
                "desc": "",
                "drgb": "#ff4f81",
                "ef": 0,
                "effect": "0",
                "grgb": "#ff4f81",
                "gt": "2",
                "gx": 2,
                "himg": "https://staticlive.douyucdn.cn/upload/dygift/1707/35f0d563198103e0be2d53e3a76f0d45.gif",
                "id": "989",
                "intro": "\u840c\u840c\u7684\u732b\u8033\u5934\u9970\uff0c\u6234\u4e0a\u4e00\u79d2\u53d8\u840c\u59b9",
                "m_ef_gif_1": "",
                "m_ef_gif_2": "",
                "mgif": "https://staticlive.douyucdn.cn/upload/dygift/1707/60a6f955eb4efd2ccdcefe652af0b52c.gif",
                "mimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/182479ede0aee81782b8ef01890baa36.png",
                "mobile_big_effect_icon_0": "",
                "mobile_big_effect_icon_1": "",
                "mobile_big_effect_icon_2": "",
                "mobile_big_effect_icon_3": "",
                "mobile_icon_v2": "https://staticlive.douyucdn.cn/upload/dygift/1707/c3fae577107f0ca78fa5a8b039f6c4b7.png",
                "mobile_small_effect_icon": "",
                "mobile_stay_time": "4000",
                "mobimg": "",
                "name": "\u732b\u8033",
                "pad_big_effect_icon": "",
                "pc": "20",
                "pdbimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/370fcfee418f752d5482a4528ae9fcdc.png",
                "pdhimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/667a86e78151e18fc820a8556ac41223.gif",
                "pimg": "https://staticlive.douyucdn.cn/upload/dygift/1707/6be027bb0d22e94650550b533f532fa8.png",
                "pt": "",
                "small_effect_icon": "",
                "sort": "3",
                "stay_time": 4000,
                "type": "2",
                "urgb": "#ff4f81"
            },
            {
                "big_effect_icon": "",
                "brgb": "#f68500",
                "ch": "0",
                "cimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/a07b16f0408e2510dae84581c8e23949.gif",
                "desc": "",
                "drgb": "#f68500",
                "ef": 0,
                "effect": "0",
                "grgb": "#f68500",
                "gt": "0",
                "gx": 1,
                "himg": "https://staticlive.douyucdn.cn/upload/dygift/1704/773478f5b8e40c141d0502aad78cc39e.gif",
                "id": "754",
                "intro": "\u62ff\u53bb\u5427\uff01\u6c34\u624b\u670d\uff01",
                "m_ef_gif_1": "",
                "m_ef_gif_2": "",
                "mgif": "https://staticlive.douyucdn.cn/upload/dygift/1704/f10a849cf1a7195c6d95d3040be5f820.png",
                "mimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/218e53cb3a7262056310a7b1ed191538.png",
                "mobile_big_effect_icon_0": "",
                "mobile_big_effect_icon_1": "",
                "mobile_big_effect_icon_2": "",
                "mobile_big_effect_icon_3": "",
                "mobile_icon_v2": "https://staticlive.douyucdn.cn/upload/dygift/1704/8bc78d6d5d0b715875884417eabdb4d6.png",
                "mobile_small_effect_icon": "",
                "mobile_stay_time": "2500",
                "mobimg": "",
                "name": "\u6c34\u624b\u670d",
                "pad_big_effect_icon": "",
                "pc": "10",
                "pdbimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/7c93d2b5ada217a00a439d54bf4682ad.png",
                "pdhimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/03431ef6752f93769eb3ec3055a17a9a.gif",
                "pimg": "https://staticlive.douyucdn.cn/upload/dygift/1704/e62c41b255da3b37779e2e327c0e96b8.png",
                "pt": "",
                "small_effect_icon": "",
                "sort": "2",
                "stay_time": 2000,
                "type": "2",
                "urgb": "#f68500"
            },
            {
                "big_effect_icon": "",
                "brgb": "#fdd592",
                "ch": "0",
                "cimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/c3226805223b413ff84fa972b4ebdb13.png",
                "desc": "",
                "drgb": "#210101",
                "ef": 0,
                "effect": "0",
                "grgb": "#c64b00",
                "gt": "0",
                "gx": 1,
                "himg": "https://staticlive.douyucdn.cn/upload/dygift/1606/072dc9aea67fb42f095f973c8e0a183e.png",
                "id": "191",
                "intro": "\u7edd\u4f73\u7684\u589e\u91cd\u7f8e\u98df",
                "m_ef_gif_1": "",
                "m_ef_gif_2": "",
                "mgif": "",
                "mimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/37a913eb1d9b3e8e6278d8a309696d58.png",
                "mobile_big_effect_icon_0": "",
                "mobile_big_effect_icon_1": "",
                "mobile_big_effect_icon_2": "",
                "mobile_big_effect_icon_3": "",
                "mobile_icon_v2": "https://staticlive.douyucdn.cn/upload/dygift/1606/96f05ad1f8b1eb5bb37526533c9d6bfb.png",
                "mobile_small_effect_icon": "",
                "mobile_stay_time": "1500",
                "mobimg": "",
                "name": "100\u9c7c\u4e38",
                "pad_big_effect_icon": "",
                "pc": "100",
                "pdbimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/9cae286a5687e8709c6fe957bea41c06.png",
                "pdhimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/af31b1e87d5ce95abb9e63d2cb73b821.png",
                "pimg": "https://staticlive.douyucdn.cn/upload/dygift/1606/9251ec1e5badb67527b99b15e1f96a95.png",
                "pt": "",
                "small_effect_icon": "",
                "sort": "1",
                "stay_time": 1000,
                "type": "1",
                "urgb": "#dd4e00"
            }
        ],
        "gift_ver": "0",
        "hls_url": "http://hls3.douyucdn.cn/live/2020877rvPdqu4JZ/playlist.m3u8?wsSecret=313a1a176b54c1509f7be402908dca49&wsTime=1500653523",
        "isVertical": 0,
        "is_pass_player": 0,
        "is_white_list": "0",
        "low_credit": "4",
        "nickname": "\u7eb3\u8c46nado",
        "online": 290452,
        "owner_avatar": "http://uc.douyutv.com/upload/avanew/face/201706/08/07/a7a8ff739cd49680b8eb7959f0be6790_big.jpg",
        "owner_uid": "128027279",
        "owner_weight": "23.2t",
        "p2p_settings": {
            "m_dm": 0,
            "player": 0,
            "use_p2p": 0,
            "w_dm": 0
        },
        "room_dm_delay": 90,
        "room_id": 2020877,
        "room_name": "\u8c46\u5b50\uff1a\u563b\u563b\u563b\u563b\u563b\u563b\u3002",
        "room_src": "https://rpic.douyucdn.cn/a1707/21/23/2020877_170721235923.jpg",
        "rtmp_cdn": "ws",
        "rtmp_live": "2020877rvPdqu4JZ.flv?wsAuth=8aa586c6d4775189b581e67f4ca3ebe5&token=app-androidhd1-0-2020877-43038f5ffef96fb3f31d40f85091cd3d&logo=0&expire=0",
        "rtmp_multi_bitrate": {
            "middle": "2020877rvPdqu4JZ_550.flv?wsAuth=7777c915fe695ab44f0d2ed3338639ec&token=app-androidhd1-0-2020877-da2b1b6fe1533e23e8440654c8c04e4c&logo=0&expire=0",
            "middle2": "2020877rvPdqu4JZ_900.flv?wsAuth=0cc3d3b0650f547c511e2e0632b93dec&token=app-androidhd1-0-2020877-9db9bd7bd9dd9c99926d17355a987b5c&logo=0&expire=0"
        },
        "rtmp_url": "http://hdl3.douyucdn.cn/live",
        "show_details": "\u76f4\u64ad\u901a\u77e5\u7fa4\uff08\u5168\u5458\u7981\u8a00\uff09\uff1a549194877\n\u8c46\u9c7c\u8bad\u7ec3\u8425\uff08\u804a\u5929\u7fa4\uff09\uff1a651127067\n\u7f51\u6613\u4e91\uff1a\u8c46\u513fcoral\n\u7f51\u6613\u4e91\u7535\u53f0\uff1a\u5927\u5200\u7d20\u8c46\u306e\u5408\u96c6\nyy\u5e38\u5e74\uff1a12960326\n\u7136\u540e\u4e0d\u77e5\u9053\u8981\u5199\u5565\u4e86",
        "show_status": "1",
        "show_time": "1500638778",
        "specific_catalog": "nado",
        "specific_status": "1",
        "url": "/nado",
        "use_p2p": "0",
        "vertical_src": "https://rpic.douyucdn.cn/a1707/21/23/2020877_170721235923.jpg",
        "vod_quality": "0"
    },
    "error": 0
}
```

File: src/streamlink/plugins/douyutv.py (hunk 1 above)
Comment (beardypig) [inline src/streamlink/plugins/douyutv.py]: If they are equivalent they should be both named the same - ie. Rename middle2 to middle (or medium). :-)

Comment (steven7851) [inline src/streamlink/plugins/douyutv.py]: They are not equivalent.
As ```rtmp_multi_bitrate:{middle: "_550", middle2: "_900"}```,
we can rename ```middle2=medium```, ```middle=low```.

**Comment by gravyboat** [comment]
thanks for the update @fozzysec, I'm fine with this now.

**Pull request merged**